test/spec/repeat.spec.js

```diff
@@ -70,6 +70,26 @@ describe( 'repeat functionality', () => {
             repeatButton.click();
             expect( [ ...form.view.html.querySelectorAll( '[name="/repdef/rep/num"]' ) ].map( i => i.value ) ).toEqual( [ '5', '5', '5' ] );
         } );
+
+        // https://github.com/enketo/enketo-core/issues/720
+        it( 'nested repeats are added correctly', () => {
+            const form = loadForm( 'nested-repeats.xml');
+            form.init();
+
+            // add repeats by clicking the add buttons
+            form.view.$.find( '.or-repeat-info[data-name="/data/repeat1/repeat11/repeat111"] .add-repeat-btn' ).click();
+            form.view.$.find( '.or-repeat-info[data-name="/data/repeat1/repeat11"] .add-repeat-btn' ).click();
+            form.view.$.find( '.or-repeat-info[data-name="/data/repeat1"] .add-repeat-btn' ).click();
+
+            expect(
+                form.getDataStr().replace( />\s+</g, '><' )
+            ).toContain( '<repeat1><repeat11><repeat111><room_open_stat/><school_roomtype/></repeat111><repeat111><room_open_stat/><school_roomtype/></repeat111></repeat11><repeat11><repeat111><room_open_stat/><school_roomtype/></repeat111></repeat11></repeat1><repeat1><repeat11><repeat111><room_open_stat/><school_roomtype/></repeat111></repeat11></repeat1>' );
```

Comment:
So in the end after your investigation found that there is no issue when loading a record, right? The issue was with the creation of the **original record** instead, right?

If not, please add a test that reproduces the issue when loading a record (using the same form).

Reply:
yes, it seems like this only occurs with creation of a new record or while modifying an existing record by adding nested repeats